File: pgm/ABC_Visual_Basic/ABC_352_A.vb
' Option Strict On
' Imports System.Collections.Generic
' Imports System.Globalization
' Imports System.IO
' Imports System.Linq
' Imports System.Text
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Sub Main()
        Dim arr() As String = Console.ReadLine().Split()
        Const one As Integer = 1
        Const two As Integer = 2
        Const three As Integer = 3
        Const zero As Integer = 0
        Dim N As Integer = Integer.Parse(arr(zero))
        Dim X As Integer = Integer.Parse(arr(one))
        Dim Y As Integer = Integer.Parse(arr(two))
        Dim Z As Integer = Integer.Parse(arr(three))
        Const Yes As String = "Yes"
        Const No As String = "No"
        If Math.Min(X, Y) <= Z AndAlso Z <= Math.Max(X, Y) Then
            Console.WriteLine(Yes)
        Else
            Console.WriteLine(No)
        End If
        Return
    End Sub
End Class

File: pgm/ABC_Dart/ABC_352_A.dart
// import 'dart:collection';
import 'dart:io';
import 'dart:math';
// import 'dart:typed_data';

void main() {
  final [int _, int X, int Y, int Z] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const String Yes = 'Yes';
  const String No = 'No';
  print(min(X, Y) < Z && Z < max(X, Y) ? Yes : No);
  exit(0);
}
